src/coreclr/nativeaot/Runtime/gcrhenv.cpp

```diff
@@ -659,10 +659,48 @@ void GCToEEInterface::SyncBlockCacheWeakPtrScan(HANDLESCANPROC /*scanProc*/, uin
 
 void GCToEEInterface::SyncBlockCacheDemote(int /*max_gen*/)
 {
+    int condemned = GCHeapUtilities::GetGCHeap()->GetCondemnedGeneration();
+
+    FOREACH_THREAD(pThread)
+    {
+        int32_t generation = pThread->GetGeneration();
+
+        // the stack is too old to be interesing in this GC
+        if (generation > condemned)
+            continue;
+
+        // the stack is as young as it can be
+        if (generation == 0)
+            continue;
+
+        pThread->SetGeneration(0);
```

Comment:
Just setting generation to 0 as we do not want to dig through the entire stack and see what aged and what did not.